_rules/visible-label-in-accessible-name-2ee8b8.md

```diff
@@ -38,9 +38,7 @@ This rule applies to any element that has:
 
 ## Expectation
 
-The complete [visible text content][] of the target element either matches or is contained within its [accessible name][].
-
-**Note:** Leading and trailing [whitespace][] and difference in case sensitivity should be ignored.
+All [text nodes][] in the [visible text content][] of each target element matches, or is contained within the [accessible name][] of the target element, except for characters used to express [non-text content][]. Leading and trailing [whitespace][] and difference in case sensitivity should be ignored.
```

Comment:
```suggestion
Each [text nodes][] in the [visible text content][] of each target element matches, or is contained within the [accessible name][] of this target element, except for characters used to express [non-text content][]. Leading and trailing [whitespace][] and difference in case sensitivity should be ignored.
```

Reply:
Subtle but (IMO) important distinction here. The expectation applies to **each** applicable element, but within that element, **all** text nodes have to meet the condition or follow the exception.

Reply:
The question is whether we want:
* each individual text node matches or is contained.
* all text nodes, grouped together in one single thingie, match or are contained.

I am not sure what it means for "all test nodes" to match a string. We need to somehow gather "all text nodes" into a single stuff that can match the string. Is it concatenation in tree order? And then, the trimming of whitespace, does it occur before or after gathering all text nodes into a single thingie?

Maybe we want to call in @EmmaJP 😄

Reply:
I've tried to adjust the order of the wording. Hope that helps clarify it.